json_container/inc/leatherman/json_container/json_container.hpp

```diff
@@ -7,17 +7,14 @@
 #include <typeinfo>
 #include <memory>
 
-// Forward declarations for rapidjson
+// The full definition of RapidJSON::GenericValue is needed for the definition
+// of the getWithDefault function in this header.
+#include <rapidjson/document.h>
```

Comment:
This is the first time we require rapidjson headers in the json_container header, so I take it back, this looks intentional.

I think we could preserve only needing forward-declarations by wrapping all the jval manipulation in another function, and if it returns a nullptr then return the default.